test/document_stores/test_faiss.py

```diff
@@ -8,294 +9,266 @@
-    faiss_index.nprobe = 2
-    document_store = FAISSDocumentStore(
-        sql_url=f"sqlite:////{tmp_path/'haystack_test_faiss.db'}",
-        faiss_index=faiss_index,
-        index=index,
-        isolation_level="AUTOCOMMIT",
-    )
+        retriever = EmbeddingRetriever(
+            document_store=document_store, embedding_model="nreimers/albert-small-v2", use_gpu=False
+        )
+        result = retriever.retrieve(query="How to test this?")
+
+        assert len(result) == len(documents_embedding_only)
+        assert type(result[0]) == Document
+
+        # Cleanup
+        document_store.faiss_indexes[document_store.index].reset()
+
+    @pytest.mark.integration
+    def test_passing_index_from_outside(self, documents_embedding_only, tmp_path):
+        d = 768
+        nlist = 2
+        quantizer = faiss.IndexFlatIP(d)
+        index = "haystack_test_1"
+        faiss_index = faiss.IndexIVFFlat(quantizer, d, nlist, faiss.METRIC_INNER_PRODUCT)
+        faiss_index.set_direct_map_type(faiss.DirectMap.Hashtable)
+        faiss_index.nprobe = 2
+        document_store = FAISSDocumentStore(
+            sql_url=f"sqlite:///{tmp_path}/haystack_test_faiss.db",
+            faiss_index=faiss_index,
+            index=index,
+            isolation_level="AUTOCOMMIT",
+        )
 
-    document_store.delete_documents()
-    # as it is a IVF index we need to train it before adding docs
-    document_store.train_index(DOCUMENTS)
-
-    document_store.write_documents(documents=DOCUMENTS)
-    documents_indexed = document_store.get_all_documents()
+        document_store.delete_documents()
+        # as it is a IVF index we need to train it before adding docs
+        document_store.train_index(documents_embedding_only)
 
-    # test if vectors ids are associated with docs
-    for doc in documents_indexed:
-        assert 0 <= int(doc.meta["vector_id"]) <= 7
+        document_store.write_documents(documents=documents_embedding_only)
+        documents_indexed = document_store.get_all_documents()
 
+        # test if vectors ids are associated with docs
+        for doc in documents_indexed:
+            assert 0 <= int(doc.meta["vector_id"]) <= 7
 
-@pytest.mark.integration
-def test_pipeline_with_existing_faiss_docstore(tmp_path):
+    @pytest.mark.integration
+    def test_pipeline_with_existing_faiss_docstore(self, ds, documents_embedding_only, tmp_path):
 
-    document_store: FAISSDocumentStore = FAISSDocumentStore(
-        sql_url=f'sqlite:///{(tmp_path / "faiss_document_store.db").absolute()}'
-    )
-    retriever = MockDenseRetriever(document_store=document_store)
-    document_store.write_documents(DOCUMENTS)
-    document_store.update_embeddings(retriever=retriever, update_existing_embeddings=True)
+        retriever = MockDenseRetriever(document_store=ds)
+        ds.write_documents(documents_embedding_only)
+        ds.update_embeddings(retriever=retriever, update_existing_embeddings=True)
```

Comment:
No need to update the embeddings, the docs already have them
```suggestion
        ds.write_documents(documents_embedding_only)
```